x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/CloseJobAction.java

```diff
@@ -38,9 +42,9 @@ private CloseJobAction() {
 
         public static final ParseField TIMEOUT = new ParseField("timeout");
         public static final ParseField FORCE = new ParseField("force");
-        @Deprecated
-        public static final String ALLOW_NO_JOBS = "allow_no_jobs";
-        public static final ParseField ALLOW_NO_MATCH = new ParseField("allow_no_match", ALLOW_NO_JOBS);
+        public static final ParseField ALLOW_NO_MATCH = new ParseField("allow_no_match").forRestApiVersion(equalTo(RestApiVersion.V_8));
```

Comment:
`allow_no_match` is the new preferred field, right?
I guess we want this to work for all future versions. You can consider using `onOrAfter` instead of `equalTo` 
in previous comment I shared a link to a PR which changed `ReindexRequest` I think the link to a current version of ReindexRequest would be more accurate

https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java#L353-L366